apps/cowswap-frontend/src/legacy/components/Tokens/TokensTableRow.tsx

```diff
@@ -148,20 +140,10 @@ export const TokensTableRow = ({
           </ApproveLabel>
         </CustomLimit>
       )
-    } else if (!isApproved || hasNoAllowance) {
-      return <TableButton onClick={handleApprove}>Approve</TableButton>
-    } else {
-      return <ApproveLabel color={theme.green1}>Approved ✓</ApproveLabel>
     }
-  }, [currentAllowance, handleApprove, isApproved, isPendingApprove, hasNoAllowance, theme.green1])
 
-  useEffect(() => {
-    if (approvalState === ApprovalState.PENDING) {
-      setApproving(true)
-    } else if (prevApprovalState === ApprovalState.PENDING && approvalState === ApprovalState.NOT_APPROVED) {
-      setApproving(false)
-    }
-  }, [approvalState, prevApprovalState, approving])
+    return <CardsSpinner />
+  }, [currentAllowance, handleApprove, approvalState, theme.green1])
```

Comment:
Why does it depend on a theme color?!

Reply:
Good catch!
Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/ffa16cd075a10c4e00d5087d81f7250a9f193f61